ghcide/src/Development/IDE/LSP/HoverDefinition.hs

```diff
@@ -76,8 +77,11 @@ request
   -> LSP.LspM c (Either ResponseError b)
 request label getResults notFound found ide (TextDocumentPositionParams (TextDocumentIdentifier uri) pos) = liftIO $ do
     mbResult <- case uriToFilePath' uri of
-        Just path -> logAndRunRequest label getResults ide pos path
-        Nothing   -> pure Nothing
+        Just path
+          | takeExtension path `elem` [".hs", ".lhs"]
+          -> logAndRunRequest label getResults ide pos path
+          | otherwise -> pure Nothing
+        Nothing  -> pure Nothing
```

Comment:
This is a bit hacky.

Reply:
Yeah, this is no good. I believe @fendor had some thoughts about how to extend the `his-plugin-api` with a notion of file extension.

Reply:
> I believe @fendor had some thoughts about how to extend the his-plugin-api with a notion of file extension.

@fendor could you elaborate that, to help this make progress? thanks!